tests/unit/test_local_project.py

```diff
@@ -389,3 +390,196 @@ def test_offline_no_clone_no_temp_dir():
+        iterdir=lambda: [reference_repo],
+        joinpath=lambda x: "/reference/repo/package_name",
+    )
+    repo_cache = RepositoryCache(cache_path=cache_path_mock, add_new=False)
+
+    flexmock(git.Repo).should_receive("clone_from").with_args(
+        url="https://server.git/my_namespace/package_name",
+        reference="/reference/repo/package_name",
+        to_path="some/temp/dir",
+        tags=True,
+    ).and_return(
+        flexmock(
+            active_branch=flexmock(name="branch"),
+            working_dir="some/temp/dir",
+            head=flexmock(is_detached=False),
+        )
+    )
+
+    flexmock(tempfile).should_receive("mkdtemp").and_return("some/temp/dir")
+
+    project = LocalProject(
+        git_url="https://server.git/my_namespace/package_name",
+        repo_name="package_name",
+        git_service=flexmock()
+        .should_receive("get_project")
+        .and_return(flexmock())
+        .mock(),
+        cache=repo_cache,
+    )
+
+    assert project
+    assert project.working_dir == Path("some/temp/dir")
+    assert project.git_url == "https://server.git/my_namespace/package_name"
+    assert project.namespace == "my_namespace"
+    assert project.repo_name == "package_name"
+    assert project.git_service
+    assert project.git_project
+    assert project.git_repo
+    assert project.ref == "branch"
+
+
+def test_clone_using_cache_not_present():
+    reference_repo = flexmock(
+        is_dir=lambda: True,
+        __str__="/reference/repo/different_package",
+        name="different_package",
+    )
+    cache_path_mock = flexmock(
+        is_dir=lambda: True,
+        iterdir=lambda: [reference_repo],
+        joinpath=lambda x: "/reference/repo/package_name",
+    )
+    repo_cache = RepositoryCache(cache_path=cache_path_mock, add_new=False)
+
+    flexmock(git.Repo).should_receive("clone_from").with_args(
+        url="https://server.git/my_namespace/package_name",
+        to_path="some/temp/dir",
+        tags=True,
+    ).and_return(
+        flexmock(
+            active_branch=flexmock(name="branch"),
+            working_dir="some/temp/dir",
+            head=flexmock(is_detached=False),
+        )
+    )
+
+    flexmock(tempfile).should_receive("mkdtemp").and_return("some/temp/dir")
+
+    project = LocalProject(
+        git_url="https://server.git/my_namespace/package_name",
+        repo_name="package_name",
+        git_service=flexmock()
+        .should_receive("get_project")
+        .and_return(flexmock())
+        .mock(),
+        cache=repo_cache,
+    )
+
+    assert project
+    assert project.working_dir == Path("some/temp/dir")
+    assert project.git_url == "https://server.git/my_namespace/package_name"
+    assert project.namespace == "my_namespace"
+    assert project.repo_name == "package_name"
+    assert project.git_service
+    assert project.git_project
+    assert project.git_repo
+    assert project.ref == "branch"
+
+
+def test_clone_and_add_to_cache():
```

Comment:
another interesting test case could be to clone a repo twice with add_new=True

Reply:
Just added the requre test for that.